ang/civiawards/dashboard/directives/more-filters-dashboard-action-button.controller.js

```diff
@@ -86,31 +77,32 @@
      * Apply filter
      */
     function applyFilter () {
-      processMyAwardsFilter(model.selectedFilters.awardFilter)
-        .then(processAwardSubtypeFilters)
-        .then(function (awardSubtypeIds) {
-          var isCaseTypeActive = model.selectedFilters.onlyShowDisabledAwards
-            ? '0'
-            : '1';
-          var param = {
-            'case_type_id.is_active': isCaseTypeActive,
-            case_type_id: awardSubtypeIds.length > 0
-              ? { IN: awardSubtypeIds }
-              : { 'IS NULL': 1 }
-          };
-
-          if (model.selectedFilters.statuses.length > 0) {
-            var statusIds = Select2Utils.getSelect2Value(model.selectedFilters.statuses);
-            var statusGroupings = getGroupingsForCaseStatuses(statusIds);
-
-            param.status_id = { IN: statusIds };
-            param['status_id.grouping'] = { IN: statusGroupings };
-          } else {
-            param.status_id = { 'IS NOT NULL': 1 };
-          }
-
-          $rootScope.$broadcast('civicase::dashboard-filters::updated', param);
-        });
+      var isCaseTypeActive = model.selectedFilters.onlyShowDisabledAwards
+        ? '0'
+        : '1';
+
+      $scope.activityFilters.case_filter['case_type_id.is_active'] = isCaseTypeActive;
+
+      if (model.selectedFilters.statuses.length > 0) {
+        var statusIds = Select2Utils.getSelect2Value(model.selectedFilters.statuses);
+        var statusGroupings = getGroupingsForCaseStatuses(statusIds);
+
+        $scope.activityFilters.case_filter.status_id = { IN: statusIds };
+        $scope.activityFilters.case_filter['status_id.grouping'] = { IN: statusGroupings };
+      } else {
+        delete $scope.activityFilters.case_filter['status_id.grouping'];
+        $scope.activityFilters.case_filter.status_id = { 'IS NOT NULL': 1 };
+      }
+
+      if (model.selectedFilters.managed_by !== 'all_awards') {
+        $scope.activityFilters.case_filter['case_type_id.managed_by'] = model.selectedFilters.managed_by;
+      } else {
+        delete $scope.activityFilters.case_filter['case_type_id.managed_by'];
+      }
+
+      $scope.activityFilters.case_filter['case_type_id.award_detail_params'] = getAwardApiFilters();
+
+      $rootScope.$broadcast('civicase::dashboard-filters::updated');
```

Comment:
@deb1990 isn't this updating the filters and triggering multiple requests?

I thought we were going to include this into a service that can be requested by the dashboard directive before it executed the request to fetch the workflows.

Reply:
No its not, because previously the initial logic to Apply My Awards was being applied using this Controller. But now it does not do anything initially.
Now https://github.com/compucorp/uk.co.compucorp.civiawards/pull/177/files#diff-64f7e3308246b806bc6b1307306016ee6c0acd0c249fd17763e9cd63f3228391R35 does that. That way we dont have to make multiple calls.

> I thought we were going to include this into a service that can be requested by the dashboard directive before it executed the request to fetch the workflows.

This was tough to implement, as currently we a controller for every dashboard action button, and not a service. So there is no way to apply functions before init without refactoring the code a lot.